### PR TITLE
Implement Interface / generate record / generate DU cases light bulbs

### PR DESCRIPTION
**Pull request by vasily-kirichenko**

### CODE REVIEW

**Comment by vasily-kirichenko** [comment]
All smart tags ported to Light Bulb.

File: src/FSharpVSPowerTools.Logic/ImplementInterfaceSmartTagger.fs [hunk 1]
```diff
@@ -24,18 +24,21 @@ type InterfaceState =
...
-                                   view: ITextView, 
-                                   editorOptionsFactory: IEditorOptionsFactoryService, 
-                                   textUndoHistory: ITextUndoHistory,
-                                   vsLanguageService: VSLanguageService, 
-                                   serviceProvider: IServiceProvider,
-                                   projectFactory: ProjectFactory,
-                                   objectIdentifier: string,
-                                   defaultBody: string) as self =
-    let tagsChanged = Event<_, _>()
+type ImplementInterface
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic/ImplementInterfaceSmartTagger.fs]: Could you get rid of these warnings https://ci.appveyor.com/project/dungpa/visualfsharppowertools/build/2.1.0-a1508100652/messages?

File: src/FSharpVSPowerTools.Logic/UnionPatternMatchCaseGeneratorSmartTagger.fs [hunk 2]
```diff
@@ -25,13 +24,41 @@ type UnionPatternMatchCaseGeneratorSmartTagger
...
+
+        transaction.Complete()
+
+    let getSuggestion snapshot (patMatchExpr, entity, insertionParams) =
+        { new ISuggestion with
+              member __.Invoke() = handleGenerateUnionPatternMatchCases snapshot patMatchExpr insertionParams entity
+              member __.NeedsIcon = false
+              member __.Text = Resource.unionPatternMatchCaseCommandName }
+
+//    member x.GetSmartTagActions(snapshot, expression, insertionPos, entity: FSharpEntity) =
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Logic/UnionPatternMatchCaseGeneratorSmartTagger.fs]: Please remove this commented code block.

File: src/FSharpVSPowerTools.Logic/ImplementInterfaceSmartTagger.fs (hunk 1 above)
Comment (vasily-kirichenko) [inline src/FSharpVSPowerTools.Logic/ImplementInterfaceSmartTagger.fs]: Done.

File: src/FSharpVSPowerTools.Logic/UnionPatternMatchCaseGeneratorSmartTagger.fs (hunk 2 above)
Comment (vasily-kirichenko) [inline src/FSharpVSPowerTools.Logic/UnionPatternMatchCaseGeneratorSmartTagger.fs]: Done.

**Comment by dungpa** [comment]
Great jobs :clap:.

**Pull request merged**